sdb/commands/internal/p2.py

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright 2020 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Common C macros for bit-manipulation and alignment.
+"""
+
+import sdb
+
+
+def p2roundup_typed(val: int, align: int, type_: str) -> int:
+    """
+    Round up `val` to the next `align` boundary (value sized as `type_`).
+    """
+    typed_val = sdb.create_object(type_, val)
+    typed_align = sdb.create_object(type_, align)
+    return (((typed_val - 1) | (typed_align - 1)) + 1).value_()
```

Comment:
Are there any use cases where it's important that we do this with the specified type, vs. using python's default numbers?  If so, that seems pretty subtle and perhaps worthy of a comment.

Reply:
For some reason I thought it would but trying out a few edge-cases in the `drgn`-repl I don't think that it matters. I'll use the default python.